DOCSP-26412 Update "Insert One" Usage Example w/ Document Structs (#193)
mongodb/docs-golang: source/usage-examples/insertOne.txt

.. include:: /includes/usage-examples/run-example-tip.rst

This example uses the following ``Restaurant`` struct as a model for documents
in the ``restaurants`` collection:

.. literalinclude:: /includes/usage-examples/code-snippets/insertOne.go
   :start-after: start-restaurant-struct
   :end-before: end-restaurant-struct
   :language: go 
   :copyable: 
   :dedent:

> The ``omitempty`` :ref:`struct tags<golang-struct-tags>` omit the corresponding 
> fields from the inserted document when left empty.

Reviewer: I think this sentence sounds better referencing the singular form:

```suggestion
The ``omitempty`` :ref:`struct tag <golang-struct-tags>` omits the corresponding 
field from the inserted document if you do not define a field value.
```

> The following example inserts a document that contains a new restaurant's 
> ``name``, ``cuisine``, and ``borough``, in the ``restaurants`` collection:

Reviewer: Simplify

```suggestion
The following example inserts a new document to the ``restaurants`` collection:
```

---------------

After you run the full example, you can find the following inserted
document in the ``restaurants`` collection:

.. code-block:: json
   :copyable: false

   {
     "_id": ObjectId("..."),

>      "Name": "8282",
>      "Cuisine": "Korean",
>      "Borough": "Manhattan"

Reviewer: fields are automatically marshalled with lowercase field names

```suggestion
     "name": "8282",
     "cuisine": "Korean",
     "borough": "Manhattan"
```

Author: Thanks for catching this — this one went right past me.

   }

> For an example on how to find a document, see :ref:`golang-find-one`.

Reviewer: while you are editing this page:

```suggestion
For an example on how to find a document, see the :ref:`golang-find-one` usage example.
```
